### PR TITLE
Added possibility to share groovy script over git repositories other than workflowLibs from Jenkins

### PR DESCRIPTION
**Pull request by attrobit**
Our goal was to load our shared groovy scripts from our own git repository. I extended the GroovyShellDecorator for that. The decorator parses the pipeline script, search for a specific pattern, download the git libraries and add them to the class path. The following example show a recognized pattern:

```
//#git http://<my git url> <branch>
```

I have to add some documentation and extend the GitLoader to support git credentials, but i want to know if you would accept my changes.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 target
 work
+/nbproject/
```
Comment (jglick) [inline .gitignore]: Heh. Delete, you do not need it.

**Comment by jglick** [comment]
Interesting work, but I guess you missed #10.

**Comment by attrobit** [comment]
I saw #10 but it does not support our necessary feature of adding groovy files to the classloader. The library step are not part of it. Or how can i add groovy files from our own git repository to the groovy shell classloader?

**Comment by jglick** [comment]
> it does not support our necessary feature of adding groovy files to the classloader

Sure it does, that is the whole point.

**Comment by attrobit** [comment]
Looks great, thanks.

**Pull request closed without merging**